### PR TITLE
Update DataServlet to return JSON data

### PR DESCRIPTION
**Pull request by tblanshard**
- Updated DataServlet to return array of strings in JSON form.
- Updated script.js to handle JSON data.
- Updated index.html naming to make more sense given new purpose.

@iht @juliocc - thank you! :)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,20 @@
...
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    
+    ArrayList<String> messages = new ArrayList<String>(
```
Comment (iht) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:29]: You could also write 

```java
ArrayList<String> messages = List.of("This", "Is A", "Message");
```

**Review by iht — APPROVED** [review]
LGTM

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (tblanshard) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:29]: That's certainly a nicer way of doing it, thank you! I hadn't come across that method before. Unfortunately I don't think they introduced the List.of method until Java 9 and we've been asked to use Java 8 - but I'll definitely be using that method in my own coding from now on so thank you!

Comment (iht) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:29]: I was not aware you were bounded to Java 8. Please ignore my comment then! Thanks!

**Pull request merged**